src/signals/IncidentMap/components/FilterPanel/FilterPanel.test.tsx

```diff
@@ -0,0 +1,122 @@
+/* SPDX-License-Identifier: MPL-2.0 */
+/* Copyright (C) 2022 Gemeente Amsterdam */
+import { screen, render } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import configuration from 'shared/services/configuration/configuration'
+
+import useFetch from '../../../../hooks/useFetch'
+import {
+  get,
+  useFetchResponse,
+  fetchCategoriesResponse,
+  mockFilters,
+} from '../__test__'
+import { updateFilterCategory } from '../utils'
+import type { Props } from './FilterPanel'
+import { FilterPanel } from './FilterPanel'
+
+jest.mock('hooks/useFetch')
+
+jest.mock('../utils', () => ({
+  updateFilterCategory: jest.fn(),
+}))
+
+const mockSetFilters = jest.fn()
+const mockSetMapFilter = jest.fn()
+
+const defaultProps: Props = {
+  filters: [],
+  setFilters: mockSetFilters,
+  setMapMessage: mockSetMapFilter,
+}
+
+const renderFilterPanel = (props: Partial<Props> = {}) =>
+  render(<FilterPanel {...defaultProps} {...props} />)
+
+describe('FilterPanel', () => {
+  beforeEach(() => {
+    // get.mockReset()
+  })
+
+  it('get categories and set filters', () => {
+    const fetchResponseWithFilters = {
+      ...useFetchResponse,
+      data: fetchCategoriesResponse,
+    }
+
+    jest.mocked(useFetch).mockImplementation(() => fetchResponseWithFilters)
+    renderFilterPanel()
+
+    expect(get).toBeCalledWith(configuration.CATEGORIES_ENDPOINT)
+    expect(get).toBeCalledTimes(1)
+
+    expect(mockSetFilters).toBeCalledWith(mockFilters)
+  })
+
+  it('should render the filter panel', () => {
+    jest.mocked(useFetch).mockImplementation(() => useFetchResponse)
+    renderFilterPanel({ filters: mockFilters })
+
+    expect(
+      screen.getByRole('heading', { name: 'Filter op onderwerp' })
```

Comment:
- [ ] consider using data test attributes 

Currently a lot of tests are based on text values. Maybe we should be smarter and add data test ids that can be removed when building for production.

Reply:
If you look at the docs of ReactTestingLibrary it is not recommended to use data test id since it is against their guiding principles of testing the application the way users will use it. 
https://testing-library.com/docs/queries/about/

Reply:
hmm but what if the texts change when we'll be using a crm to manage translations?